src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -89,24 +85,18 @@ internal partial class RequestExecutionQueue
 
         public RequestExecutionQueue(
             ILspLogger logger,
-            LspWorkspaceRegistrationService lspWorkspaceRegistrationService,
-            LspMiscellaneousFilesWorkspace? lspMiscellaneousFilesWorkspace,
             IGlobalOptionService globalOptions,
             ImmutableArray<string> supportedLanguages,
-            WellKnownLspServerKinds serverKind)
+            WellKnownLspServerKinds serverKind,
+            RequestTelemetryLogger requestTelemetryLogger,
+            LspWorkspaceManager lspWorkspaceManager)
         {
             _logger = logger;
             _globalOptions = globalOptions;
             _supportedLanguages = supportedLanguages;
             _serverKind = serverKind;
-
-            // Pass the language client instance type name to the telemetry logger to ensure we can
-            // differentiate between the different C# LSP servers that have the same client name.
-            // We also don't use the language client's name property as it is a localized user facing string
-            // which is difficult to write telemetry queries for.
-            _requestTelemetryLogger = new RequestTelemetryLogger(_serverKind.ToTelemetryString());
```

Comment:
Moved these two values up to `LanguageServerTarget` since they're needed by the semantic tokens refresh listener